### DISCUSSION TITLE
How to whitelist a specific domain?

### DISCUSSION THREAD

**Discussion by UnNamed-M3** [AdguardTeam/AdGuardHome#3771] [Q&A]
Hi,

I'm running AdGuard Home on my Raspberry Pi in a Docker. 

I would very much like to know how I could whitelist specific domains? According to the KB, I could use @@||example.org^ to unblock the domain. However this doesn't work. 

I also tried adding the domains in the DNS Allowlist but while doing so I'm getting "Error: control/filtering/add_url | Couldn't fetch filter from url" error. So I think I need to host my own whitelist.txt file in order to get this done. 

Is my assumption correct? 
If so; how can I add this in the Work/Config data folder? 

Many thanks for any ideas.

Behrad

**Comment by timkgh** [comment]
You add `@@||example.org^` to **Custom filtering rules**

**Reply by UnNamed-M3** [reply]
Hi Timkgh,

I've already tried that but still some parts of the websites are being blocked by the DNS block lists which I've in place. I'm missing something but not quite sure what exactly...

**Reply by timkgh** [reply]
You may need to allow list more domains, you have to look in the logs for all domains that are blocked, a web site can use different domains for its content. Allow listing a domain in AGH just allows that domain and possibly its subdomains, it does not mean that all the content on that web site is allowed, it's different from how content blocking browser extensions work.

**Reply by UnNamed-M3** [reply]
Aaah I see... So then there is a mismatch with my interpretation about how this should work.  
Indeed I was expecting that by allowing a specific domain I would whitelist all of it's content, regardless of any blocklists. Pretty much like how AdBlock / Ghostery extension would work. 

Would this be perhaps something which the AdGuard team could pick up as a feature? ;)

**Reply by timkgh** [reply]
What you are looking for is simply not possible to do at the DNS level.

**Comment by fuckFH** [comment]
After conducting web research, it appears that there have been no improvements on this topic in 2024. It should be possible that when a user excludes a domain, no filter rules are applied when the user fetches the domain.

**Comment by Dynamic5912** [comment]
> After conducting web research, it appears that there have been no improvements on this topic in 2024. It should be possible that when a user excludes a domain, no filter rules are applied when the user fetches the domain.

How so? Example?
